### PR TITLE
chore(e2e-tests): add e2e test ci config

### PR DESCRIPTION
**Pull request by limhawjia**

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Build tests
+        run: |
+          make source
+          cd _source/1.24
+          ./build/run.sh make WHAT="test/e2e/e2e.test"
+
+      - name: Run tests
+        run: |
+          cd _source/1.24
+          ./_output/dockerized/bin/linux/amd64/e2e.test -provider=local -kubeconfig=$HOME/.kube/config -test.parallel=8 -ginkgo.skip='\[Slow\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]|PodSecurityPolicy|LoadBalancer|load.balancer|Simple.pod.should.support.exec.through.an.HTTP.proxy|subPath.should.support.existing|NFS|nfs|inline.execution.and.attach|should.be.rejected.when.no.endpoints.exist' -ginkgo.focus='.'
```
Comment (xuchen-xiaoying) [inline .github/workflows/e2e-test.yml:30]: is there rule for Feature related skip ? @limhawjia

Comment (limhawjia) [reply .github/workflows/e2e-test.yml:30]: the pipeline skips all test specs that contain `[Feature:xxx]` in its name, do we want to skip specific features instead?

Comment (xuchen-xiaoying) [reply .github/workflows/e2e-test.yml:30]: I mean how do we decide the skip rule, is there a guideline?

Comment (limhawjia) [reply .github/workflows/e2e-test.yml:30]: based on my understanding:

* `[Slow]`: tests that may take a long time to complete
* `[Disruptive]`: tests that do not self-isolate and may disrupt online production environments
* `[Flaky]`: tests that may occasionally fail due to unexpected reasons
* `[Feature:.+]`: tests for unstable features or feature gates
* `PodSecurityPolicy`: deprecated
* `LoadBalancer|loadbalancer': these tests are targeted at cloud providers providing their own LoadBalancer service implementations
* `Simple.pod.should.support.exec.through.an.HTTP.proxy`: flaky test (refer to https://github.com/kubernetes/kubernetes/issues/30489)
* `subPath.should.support.existing`: these tests are really slow (mentioned in https://github.com/kubernetes/test-infra/pull/17493#issue-611144171)
* `NFS|nfs`: flaky test (refer to https://github.com/kubernetes/test-infra/pull/22590)
* `inline.execution.and.attach`: flaky test (refer to https://github.com/kubernetes/kubernetes/issues/21138#issuecomment-184930152)
* `should.be.rejected.when.no.endpoints.exist`: refer to https://github.com/kubernetes/test-infra/pull/17988#discussion_r442285786

Comment (limhawjia) [reply .github/workflows/e2e-test.yml:30]: these current rules are taken from the PR pre-submit jobs of the official kubernetes repo, the configuration can found [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-testing/kubernetes-kind-presubmits.yaml)

**Pull request closed without merging**